test/tools/catch-analyzer/src/main/java/org/elasticsearch/test/ClassAnalyzer.java

```diff
@@ -0,0 +1,172 @@
+        this.className = className;
+        this.loader = loader;
+    }
+    
+    @Override
+    public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+        this.superName = superName;
+    }
+
+    @Override
+    public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
+        if (desc.contains("SwallowsExceptions")) {
+            suppressed = true;
+        }
+        return null;
+    }
+    
+    @Override
+    public void visitSource(String source, String debug) {
+        this.source = source;
+    }
+
+    @Override
+    public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+        // don't scan bridge methods: they may have annotations, but we won't find problems!
+        if ((access & Opcodes.ACC_BRIDGE) != 0) {
+            return null;
+        }
+        MethodAnalyzer analyzer = new MethodAnalyzer(loader, className, superName, access, name, desc, signature, exceptions);
+        analyses.put(new Method(name, desc), analyzer);
+        return analyzer;
+    }
+
+    @Override
+    public void visitEnd() {
+        // fold lambdas violations into their parent methods
+        List<Method> lambdas = new ArrayList<>();
+        analyses.forEach((method, analysis) -> {
+            for (Method lambda : analysis.lambdas) {
+                MethodAnalyzer lambdaAnalysis = analyses.get(lambda);
+                if (lambdaAnalysis != null) {
+                    analysis.violations.addAll(lambdaAnalysis.violations);
+                    lambdas.add(lambda);
+                } else {
+                    throw new AssertionError("where in the world is lambda " + lambda);
+                }
+            }
+        });
+        // now nuke them
+        analyses.keySet().removeAll(lambdas);
+    }
+    
+    /** log violations, grouping them by catch block entry line number for the method */
+    long logViolations(PrintStream stream) {
+        long violationCount = 0;
+        if (suppressed) {
+            long violations = 0;
+            for (MethodAnalyzer analysis : analyses.values()) {
+                violations += analysis.violations.size();
+            }
+            if (violations == 0) {
+                StringBuilder sb = createMessage("Wrong class annotation", null, -1);
+                addDetails(sb, new Violation(Kind.ANNOTATED_AS_SWALLOWER_BUT_HAS_NO_PROBLEMS, -1, -1));
+                stream.println(sb.toString());
+                violationCount++;
+            }
+        } else {
+            for (Map.Entry<Method,MethodAnalyzer> entry : analyses.entrySet()) {
+                Method method = entry.getKey();
+                MethodAnalyzer analysis = entry.getValue();
+                if (analysis.suppressed) {
+                    if (analysis.violations.isEmpty()) {
+                        StringBuilder sb = createMessage("Wrong method annotation", method, -1);
```

Comment:
Should this string be "Method marked as `@SwallowsException` but doesn't swallow any exceptions" or am reading this wrong?